Some fixes for issues raised by cppcheck

notspiff opened the pull request:
Just some small stuff I quickly fixed while testing #12543

notspiff committed:
changed: return non-const references in assignment operators

that's the normal thing to do.

quells sca warnings such as

[../kodi/xbmc/utils/Speed.h:68]: (style) 'CSpeed::operator=' should return 'CSpeed &'.

notspiff committed:
micro-optim: pass by reference

quells sca warning

[../kodi/xbmc/LangInfo.h:223]: (performance) Function parameter 'grouping' should be passed by reference.

Rechi commented:
@notspiff cpp files have to be updated as well, as currently function signature mismatches.

notspiff commented:
that was stupid.. only ran the analysers.

notspiff committed:
missing initializers

notspiff committed:
mark constructors explicit to avoid unexpected implicit casts

MartijnKaijser commented:
Jenkins build this please

Rechi commented on xbmc/cores/AudioEngine/Utils/AEChannelInfo.h:
```diff
@@ -32,7 +32,7 @@ class CAEChannelInfo {
 public:
   CAEChannelInfo();
-  CAEChannelInfo(const enum AEChannel* rhs);
+  explicit CAEChannelInfo(const enum AEChannel* rhs);
```
this causes build issues for windows and android

notspiff replied:
yes, i see. do you want to rely on implicit casts or should i change the code to be explicit?

Rechi replied:
I think explicit way is preferred, but its your choice.

notspiff replied:
if you ask me, you will always get the explicit code. i do not like implicit casts one bit.

notspiff commented:
your CI is .. ehm "slightly annoying"

Rechi commented:
>your CI is .. ehm "slightly annoying".
I know that already

jenkins build this please

notspiff commented:
lol. one phone call, one skype session and coding on this at the same time was apparently too much for my brain.

notspiff committed:
quell float/double and signed/unsigned mismatch errors

notspiff committed:
fix reorder warnings in VAAPI/VDPAU

notspiff committed:
remove unused assignment

notspiff committed:
explicitly give struct a constructor to quell static analyzers

notspiff committed:
check return value from fscanf

notspiff committed:
avoid comparison of unsigned to 0

as MinimumVersion is (curently) 0, the comparison is always true
so compiler will warn.

notspiff committed:
undef to avoid redefinition warnings

Rechi commented:
jenkins build this please

MartijnKaijser commented:
Only osx fail due to code change

notspiff committed:
remove unused assignment

notspiff committed:
print unsigned as unsigned

notspiff committed:
prefer c++ style casts

Rechi commented:
jenkins build this please

Rechi commented:
jenkins build this please
